### PR TITLE
Fix parameters not showing up in 'nameof' on a method

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/60812

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@genlu ptal

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs [hunk 1]
```diff
@@ -346,22 +345,38 @@ private ImmutableArray<ISymbol> GetSymbolsForExpressionOrStatementContext()
...
+                symbols = semanticModel.LookupSymbols(_context.LeftToken.SpanStart);
+
+                // We may be inside of a nameof() on a method.  In that case, we want to include the parameters in
+                // the nameof if LookupSymbols didn't already return them.
+                var enclosingMethodOrLambdaNode = contextNode.AncestorsAndSelf().FirstOrDefault(n => n is AnonymousFunctionExpressionSyntax or BaseMethodDeclarationSyntax);
+                var enclosingMethodOrLambda = enclosingMethodOrLambdaNode is null
+                    ? null
+                    : semanticModel.GetSymbolInfo(enclosingMethodOrLambdaNode).GetAnySymbol() ?? semanticModel.GetDeclaredSymbol(enclosingMethodOrLambdaNode);
+                if (enclosingMethodOrLambda is IMethodSymbol method)
+                    symbols = [.. symbols.Concat(method.Parameters)];
```
Comment (ToddGrun) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:365]: >[..  [](http://example.com/codeflow?start=30&length=4)

is this the same as 

symbols = symbols.Concat(method.Parameters);

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:365]: likely yes.  will fixup.

Comment (ToddGrun) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:365]: That's confusing. symbols is an IA, method.Parameters is an IA, isn't this invoking the IA overload for Concat?

Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:365]: no, it's not.  i get an error when i try :)

Comment (ToddGrun) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:365]: oh, I see. One is IA<ISymbol>, the other is IA<IParamaterSymbol>. It needs to be:

symbols.Concat(method.Parameters.Cast<ISymbol>());

Comment (ToddGrun) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:365]: oh, just a sec, I thought that cast call just cast the array, not enumerate

Comment (ToddGrun) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:365]: Duh, it's CastArray, not Cast

**Pull request merged**